pyodide-build/pyodide_build/config.py

```diff
@@ -151,6 +176,16 @@ def to_env(self) -> dict[str, str]:
 
 BUILD_VAR_TO_KEY = {v: k for k, v in BUILD_KEY_TO_VAR.items()}
 
+# Configuration keys that can be overridden by the user.
+# TODO: distinguish between variables that are overridable by the user and those that are not.
+OVERRIDABLE_BUILD_KEYS = {
+    "cflags",
+    "cxxflags",
+    "ldflags",
+    "rust_toolchain",
+    "meson_cross_file",
+}
```

Comment:
You'll need to change the name of the variable

here:

https://github.com/pyodide/pyodide/blob/43cce6e5fffa4cfcb0ae3709a9c40844e720dabb/pyodide-build/pyodide_build/config.py#L134-L136

and here

https://github.com/pyodide/pyodide/blob/43cce6e5fffa4cfcb0ae3709a9c40844e720dabb/pyodide-build/pyodide_build/config.py#L174-L176

as well, so that the `cflags` key can affect the `SIDE_MODULE_CFLAGS` env variable.

(We may change the env variable name too, but it will be a breaking change so let's to that in a separate PR).

Reply:
oh right, forgot that these were all somewhat coupled to each other.